### ISSUE TITLE
There is no way to specify `artifactType` with `oras push` cli command

### ISSUE THREAD

**Issue by asilverman** [oras-project/oras#1069]
### What happened in your environment?

I would like the manifest in my ACR instance to be equal to the following:
```json
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.ms.bicep.provider.artifact",
  "config": {
    "mediaType": "application/vnd.ms.bicep.provider.config.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
  "layers": [
    {
      "mediaType": "application/vnd.ms.bicep.provider.layer.v1.tar+gzip",
      "digest": "sha256:f1f31b47972a60b1905f3252a4819dad964d505cf530a9cd3d67cee8a3c295f5",
      "size": 9346232
    }
  ],
  "annotations": {
    "bicep.serialization.format": "v1",
    "org.opencontainers.image.created": "2023-05-04T16:40:05Z"
  }
}
```

I tried issuing the following command using `oras` 
```pwsh
oras push asilvermantestbr.azurecr.io/bicep/providers/az:ariTest `
> --annotation bicep.serialization.format=v1 `
> --artifact-type application/vnd.ms.bicep.provider.artifact `
> types.tgz:application/vnd.ms.bicep.provider.types.layer.v1.tar+gzip
Exists    f1f31b47972a types.tgz
Pushed [registry] asilvermantestbr.azurecr.io/bicep/providers/az:ariTest
Digest: sha256:ad6c4ea64efc5ce447703450af924cc4b9aa43c67017dc3c884e0022db7c71fe
```

### What did you expect to happen?

I expected to get the manifest above but got the following instead. 

Notice the following differences:
- `artifactType` is missing, instead the value for `@config.mediaType` has the value of  `--artifactType`

```pwsh
oras manifest fetch asilvermantestbr.azurecr.io/bicep/providers/az:ariTest --pretty
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "config": {
    "mediaType": "application/vnd.ms.bicep.provider.artifact",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
  "layers": [
    {
      "mediaType": "application/vnd.ms.bicep.provider.types.layer.v1.tar+gzip",
      "digest": "sha256:f1f31b47972a60b1905f3252a4819dad964d505cf530a9cd3d67cee8a3c295f5",
      "size": 9346232,
      "annotations": {
        "org.opencontainers.image.title": "types.tgz"
      }
    }
  ],
  "annotations": {
    "bicep.serialization.format": "v1",
    "org.opencontainers.image.created": "2023-08-21T21:49:41Z"
  }
}
```

### How can we reproduce it?

Follow the commands described above to reproduce the issue

### What is the version of your ORAS CLI?

Version:        1.0.1
Go version:     go1.20.7
Git commit:     acf147b7564f44f8637dc59937d6167e5427e612
Git tree state: clean

### What is your OS environment?

Win11

### Are you willing to submit PRs to fix it?

- [ ] Yes, I am willing to fix it.

**Comment by qweeah** [comment]
Below manifest has `artifactType` and customized config datat, which requires image-spec rc.4 support. oras 1.0.1 only implements image-spec rc.2, not image-spec rc.4.
```json
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.ms.bicep.provider.artifact",
  "config": {
    "mediaType": "application/vnd.ms.bicep.provider.config.v1+json",
    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
    "size": 2
  },
...
}
```

We will introduce the support in oras 1.1.0-rc.2 release. You can also try build from main branch to test it like below:
```console
# Prepare
> go install oras.land/oras/cmd/oras@main
> oras version
Version:    1.1.0-rc.1+unreleased
Go version: go1.20
> echo 123 > types.tgz # test layer
> echo {} > test.config # test config

# Push
> oras push localhost:5000/test:artifact-type types.tgz:application/vnd.ms.bicep.provider.layer.v1.tar+gzip --artifact-type application/vnd.ms.bicep.provider.artifact --config test.config:application/vnd.ms.bicep.provider.config.v1+json -a bicep.serialization.format=v1
Uploading de7488d37885 types.tgz
Uploaded  de7488d37885 types.tgz
Pushed [registry] localhost:5000/test:artifact-type
Digest: sha256:5969c3acc05be88122e3a5c17c5c9bc8ed699f6b6a3ca95c5e86bbcdcade45be

# Validate
> oras manifest fetch localhost:5000/test:artifact-type --pretty
{
  "schemaVersion": 2,
  "mediaType": "application/vnd.oci.image.manifest.v1+json",
  "artifactType": "application/vnd.ms.bicep.provider.artifact",
  "config": {
    "mediaType": "application/vnd.ms.bicep.provider.config.v1+json",
    "digest": "sha256:2e08d1f6000aef541797d008c05ac36f4dbebfb36cbac5615788e6fcc5b300a7",
    "size": 6
  },
  "layers": [
    {
      "mediaType": "application/vnd.ms.bicep.provider.layer.v1.tar+gzip",
      "digest": "sha256:de7488d3788570318ca4e8ab9f8de39221be6d38c0fa0267bfb9f943338f9363",
      "size": 12,
      "annotations": {
        "org.opencontainers.image.title": "types.tgz"
      }
    }
  ],
  "annotations": {
    "bicep.serialization.format": "v1",
    "org.opencontainers.image.created": "2023-05-04T16:40:05Z"
  }
}

**Comment by asilverman** [comment]
Thank you for the explanation, @qweeah why is config size == 6?

**Comment by qweeah** [comment]
> why is config size == 6?

Well because it's windows PowerShell :) 
1) `{}` literally means no content since I am using PowerShell
2) The default encoding output of `echo` on windows is UTF16 LE, so each character takes two bytes
3) So 6 = 2(BOM) + 0(no content) + 2(CR) + 2(LF)

Just using `echo` to generate for demo and it's not recommended for prod usage.

**Comment by yizha1** [comment]
@asilverman would you mind confirming whether this issue is solved by ORAS CLI `v1.1.0-rc.2`?

**Comment by asilverman** [comment]
Confirmed this works with version 1.1.0
